Library/Homebrew/attestation.rb

```diff
@@ -45,13 +45,13 @@ class GhAuthNeeded < RuntimeError; end
     # @api private
     sig { returns(T::Boolean) }
     def self.enabled?
-      # TODO: allow this undocumented variable until this is rolled out more
-      #       widely and then we can remove or document it.
-      return false if ENV.fetch("HOMEBREW_NO_VERIFY_ATTESTATIONS", false)
+      return false if Homebrew::EnvConfig.no_verify_attestations?
       return true if Homebrew::EnvConfig.verify_attestations?
+      return false if GitHub::API.credentials.blank?
       return false if ENV.fetch("CI", false)
+      return false unless Formula["gh"].any_version_installed?
```

Comment:
Technically we can also use `gh` when installed inside of GitHub Actions runner environments, so it might make sense to loosen this to "`gh` installed on `$PATH`" not just via Homebrew 🙂

Reply:
@woodruffw The PATH logic is non-trivial enough (and we don't provide a nice helper for this) I'm tempted to punt on it for now. Note that this will only take effect for the cases below this check, not those above (e.g. being explicitly enabled like in https://github.com/Homebrew/homebrew-core/pull/177326)